Typescript issue

Hi,

thanks for your component, it's great to use :)

Yet I have an issue with Typescript that I could not really solve.

I get the following Typescript error:

> Type 'Countdown' is missing the following properties from type 'ElementClass': context, setState, forceUpdate, props, and 2 more.

The complete message:

```
src/Countdown.tsx:27:5 - error TS2607: JSX element class does not support attributes because it does not have a 'props' property.

27     <Countdown date={date} renderer={renderCountdown} />
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/Countdown.tsx:27:6 - error TS2786: 'Countdown' cannot be used as a JSX component.
  Its instance type 'Countdown' is not a valid JSX element.
    Type 'Countdown' is missing the following properties from type 'ElementClass': context, setState, forceUpdate, props, and 2 more.

27     <Countdown date={date} renderer={renderCountdown} />
        ~~~~~~~~~

[11:33:22 AM] Found 2 errors. Watching for file changes.
```

That's basically my file:

```
import * as React from 'react'
import Countdown, { zeroPad } from 'react-countdown'

const renderCountdown = ({ days, hours, minutes, seconds, completed }) => {
    if (completed) {
        return <div>You are good to go!</div>
    } else {
        return (
            <div>
                <div>{zeroPad(days)}</div>
                <div>:</div>
                <div>{zeroPad(hours)}</div>
                <div>:</div>
                <div>{zeroPad(minutes)}</div>
                <div>:</div>
                <div>{zeroPad(seconds)}</div>
            </div>
        )
    }
}

interface IProps {
    date: string
}

const CountdownWrapper = ({ date }: IProps) => (
    <Countdown date={date} renderer={renderCountdown} />
)

export default CountdownWrapper
```

Does anybody else have the issue or know how to solve it?

NOTE: I got rid of the message by the following hack: `const CountdownWithoutTyping = Countdown as any`

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,0 +1,43 @@
+# Contributing
+
+Thanks for your interest in contributing to this project!
+
+Pull requests are very welcome to either fix existing bugs or add new features. Before submitting a pull request, please consider filing an issue to discuss the change's details.
+
+## Bug Reports
+
+1. Please [search for any similar issues](https://github.com/ndresx/react-countdown/issues?q=is%3Aissue) before as there is a chance that someone may have reported it already.
+
+2. Provide a demo of the bug in isolation if possible (e.g., codesandbox.io); otherwise, try to be as detailed as possible in the description.
+
+## Feature Requests
+
+Feature requests and new suggestions are also always welcome, but please take a moment to find out whether your idea fits within the scope of this project or not.
+
+## Pull Requests
+
+1. [Fork](https://github.com/ndresx/react-countdown/fork) the repository.
+
+2. If you are adding new functionality, or fixing a bug, provide tests with code coverage of preferably 100%.
+
+3. Push the changes to your fork and [submit a pull request](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)!
+
+## Development
+
+Once you have forked the project and checked it out locally, create a new branch from `master` and start developing!
+
+### Linting
+
+You can run the linting tools with the following command:
+
+```sh
+yarn lint
+```
+
+### Tests
+
+Make sure that all tests pass and that the code coverage is close to 100%.
+
+```sh
+yarn test
+```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -2,17 +2,17 @@
 
 Thanks for your interest in contributing to this project!
 
-Pull requests are very welcome to either fix existing bugs or add new features. Before submitting a pull request, please consider filing an issue to discuss the change's details.
+Pull requests are very welcome to either fix existing bugs, add new features, or improve the code in general. Before submitting a pull request, please consider filing an issue to discuss the change's details.
 
 ## Bug Reports
 
-1. Please [search for any similar issues](https://github.com/ndresx/react-countdown/issues?q=is%3Aissue) before as there is a chance that someone may have reported it already.
+1. Please [search for similar issues](https://github.com/ndresx/react-countdown/issues?q=is%3Aissue) before as there is a chance that someone might have reported it already.
 
 2. Provide a demo of the bug in isolation if possible (e.g., codesandbox.io); otherwise, try to be as detailed as possible in the description.
 
 ## Feature Requests
 
-Feature requests and new suggestions are also always welcome, but please take a moment to find out whether your idea fits within the scope of this project or not.
+Feature requests, improvements, and new suggestions are also always welcome, but please take a moment to find out whether your idea fits within the scope of this project or not.
 
 ## Pull Requests
 
@@ -36,7 +36,7 @@ yarn lint
 
 ### Tests
 
-Make sure that all tests pass and that the code coverage is close to 100%.
+Make sure that all tests are passing and that the code coverage is close to 100%.
 
 ```sh
 yarn test

--- a/README.md
+++ b/README.md
@@ -297,7 +297,7 @@ The `options` object consists of the following three component props and is used
 
 ## Contributing
 
-Contributions are warmly welcmoe
+Contributions of any kind are very welcome. Read more in our [contributing guide](https://github.com/ndresx/react-countdown/blob/master/CONTRIBUTING.md) about how to report bugs, create pull requests, and other development-related topics.
 
 ## License
 

--- a/README.md
+++ b/README.md
@@ -151,7 +151,7 @@ Valid values can be _(and more)_:
 * `new Date(1580518923000)` // [`Date`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object
 
 ### `key`
-This is one of React's internal component props and is used to identify the component. However, we can leverage this behavior and use it to, for example,  restart the countdown by
+This is one of React's internal component props and is used to identify the component. However, we can leverage this behavior and use it to, for example, restart the countdown by
 passing in a new `string` or `number`.
 
 Please see [official React docs](https://reactjs.org/docs/lists-and-keys.html#keys) for more information about keys.
@@ -204,7 +204,7 @@ If the current date and time (determined via a reference to `Date.now`) is not t
 `onMount` is a callback and triggered when the countdown mounts. It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ### `onStart`
-`onStart` is a callback and triggered whenever the countdown is started (including first-run).  It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
+`onStart` is a callback and triggered whenever the countdown is started (including first-run). It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ### `onPause`
 `onPause` is a callback and triggered every time the countdown is paused. It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).

--- a/README.md
+++ b/README.md
@@ -7,17 +7,18 @@ A customizable countdown component for React.
 * [Props](#props)
 * [API Reference](#api-reference)
 * [Helpers](#helpers)
+* [Contributing](#contributing)
 * [License](#license)
 
 ## Getting Started
 
-You can either install the module via `npm` or `yarn`:
+You can install the module via `npm` or `yarn`:
 
-```
+```sh
 npm install react-countdown --save
 ```
 
-```
+```sh
 yarn add react-countdown
 ```
 
@@ -142,7 +143,7 @@ ReactDOM.render(
 |[**onComplete**](#oncomplete)|`function`|`undefined`|Callback when countdown ends|
 
 ### `date`
-The `date` prop is the only required one and can be a [`Date`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object, `string`, or timestamp in the future. By default, this date value gets compared with the current date, or a custom handler defined via [`now`](#now).
+The `date` prop is the only required one and can be a [`Date`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) object, `string`, or timestamp in the future. By default, this date is compared with the current date, or a custom handler defined via [`now`](#now).
 
 Valid values can be _(and more)_:
 * `'2020-02-01T01:02:03'` // [`Date` time string format](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#Date_Time_String_Format)
@@ -187,7 +188,7 @@ _Please note that once a custom `renderer` is defined, the [`children`](#childre
 <a name="renderer"></a>
 ### `renderer(props)`
 The component's render output is very simple and depends on [`daysInHours`](#daysinhours): _{days}:{hours}:{minutes}:{seconds}_.
-If this doesn't fit your needs, a custom `renderer` callback can be defined to return a new React element. It receives an argument which consists of a time delta object (incl. `formatted` values) to help building your own representation of the countdown.
+If this doesn't fit your needs, a custom `renderer` callback can be defined to return a new React element. It receives an argument that consists of a time delta object (incl. `formatted` values) to build your own representation of the countdown.
 ```js
 { total, days, hours, minutes, seconds, milliseconds, completed }
 ```
@@ -197,22 +198,22 @@ The render props also contain the countdown's [`API`](#api-reference) as `api` p
 _Please note that once a custom `renderer` is defined, the [`children`](#children) prop will be ignored._
 
 ### `now`
-If the current datetime (determined via a reference to `Date.now`) is not the right thing to compare with for you, a reference to a custom function which returns a similar dynamic value could be provided as an alternative.
+If the current date and time (determined via a reference to `Date.now`) is not the right thing to compare with for you, a reference to a custom function that returns a similar dynamic value could be provided as an alternative.
 
 ### `onMount`
-`onMount` is a callback and triggered when the countdown mounts. It receives the time delta object which is returned by [`calcTimeDelta`](#calctimedelta).
+`onMount` is a callback and triggered when the countdown mounts. It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ### `onStart`
-`onStart` is a callback and triggered whenever the countdown is started (including first-run).  It receives the time delta object which is returned by [`calcTimeDelta`](#calctimedelta).
+`onStart` is a callback and triggered whenever the countdown is started (including first-run).  It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ### `onPause`
-`onPause` is a callback and triggered every time the countdown is paused. It receives the time delta object which is returned by [`calcTimeDelta`](#calctimedelta).
+`onPause` is a callback and triggered every time the countdown is paused. It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ### `onTick`
-`onTick` is a callback and triggered every time a new period is started, based on what the [`intervalDelay`](#intervaldelay)'s value is. It only gets triggered when the countdown's [`controlled`](#controlled) prop is set to `false`, meaning that the countdown has full control over its interval. It receives the time delta object which is returned by [`calcTimeDelta`](#calctimedelta).
+`onTick` is a callback and triggered every time a new period is started, based on what the [`intervalDelay`](#intervaldelay)'s value is. It only gets triggered when the countdown's [`controlled`](#controlled) prop is set to `false`, meaning that the countdown has full control over its interval. It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ### `onComplete`
-`onComplete` is a callback and triggered whenever the countdown ends. In contrast to [`onTick`](#ontick), the [`onComplete`](#oncomplete) callback gets also triggered in case [`controlled`](#controlled) is set to `true`. It receives the time delta object which is returned by [`calcTimeDelta`](#calctimedelta).
+`onComplete` is a callback and triggered whenever the countdown ends. In contrast to [`onTick`](#ontick), the [`onComplete`](#oncomplete) callback gets also triggered in case [`controlled`](#controlled) is set to `true`. It receives the time delta object, which is returned by [`calcTimeDelta`](#calctimedelta).
 
 ## API Reference
 
@@ -232,14 +233,14 @@ Returns a `boolean` for whether the countdown has been completed or not.
 
 ## Helpers
 
-This module also exports 3 simple helper functions which can be utilized to build your own countdown custom [`renderer`](#renderer).
+This module also exports three simple helper functions, which can be utilized to build your own countdown custom [`renderer`](#renderer).
 
 ```js
 import Countdown, { zeroPad, calcTimeDelta, formatTimeDelta } from 'react-countdown';
 ```
 
 ### `zeroPad(value, [length = 2])`
-The `zeroPad` function works similar to other well-known pad-functions and takes 2 arguments into account. A `value` which can be a `string` or `number`, as well as a `length` parameter which defaults to `2` as you are most likely only going to use this function if you actually want to pad one of your values. Either returns a `number` if `length` equals `0`, or the zero-padded `string`.
+The `zeroPad` function works similarly to other well-known pad-functions and takes two arguments into account. A `value` which can be a `string` or `number`, as well as a `length` parameter which defaults to `2` as you are most likely only going to use this function if you actually want to pad one of your values. Either returns a `number` if `length` equals `0`, or the zero-padded `string`.
 
 ```js
 const renderer = ({ hours, minutes, seconds }) => (
@@ -257,7 +258,7 @@ const renderer = ({ hours, minutes, seconds }) => (
 { total, days, hours, minutes, seconds, milliseconds, completed }
 ```
 
-This function accepts 2 arguments in total, only the first one is required.
+This function accepts two arguments in total; only the first one is required.
 
 **`date`**
 Date or timestamp representation of the end date. See [`date`](#date) prop for more details.
@@ -283,7 +284,7 @@ Defines the offset time that gets added to the start time; only considered if co
 { days, hours, minutes, seconds }
 ```
 
-This function accepts 2 arguments in total, only the first one is required.
+This function accepts two arguments in total; only the first one is required.
 
 **`delta`**
 Time delta object, e.g.: returned by [`calcTimeDelta`](#calctimedelta).
@@ -293,6 +294,10 @@ The `options` object consists of the following three component props and is used
 * [`daysInHours`](#daysinhours)
 * [`zeroPadTime`](#zeropadtime)
 * [`zeroPadDays`](#zeropaddays)
+
+## Contributing
+
+Contributions are warmly welcmoe
 
 ## License
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-countdown",
-  "version": "2.2.1",
+  "version": "2.2.2",
   "description": "A customizable countdown component for React.",
   "main": "./dist/index.js",
   "module": "./dist/index.es.js",

--- a/src/Countdown.tsx
+++ b/src/Countdown.tsx
@@ -21,6 +21,7 @@ export interface CountdownProps
   readonly intervalDelay?: number;
   readonly precision?: number;
   readonly autoStart?: boolean;
+  readonly className?: string;
   readonly children?: React.ReactElement<any>;
   readonly renderer?: (props: CountdownRenderProps) => React.ReactNode;
   readonly now?: () => number;
@@ -29,7 +30,6 @@ export interface CountdownProps
   readonly onPause?: CountdownTimeDeltaFn;
   readonly onTick?: CountdownTimeDeltaFn;
   readonly onComplete?: CountdownTimeDeltaFn | LegacyCountdownProps['onComplete'];
-  readonly className?: string;
 }
 
 export interface CountdownRenderProps extends CountdownTimeDelta {
@@ -78,6 +78,7 @@ export default class Countdown extends React.Component<CountdownProps, Countdown
     intervalDelay: PropTypes.number,
     precision: PropTypes.number,
     autoStart: PropTypes.bool,
+    className: PropTypes.string,
     children: PropTypes.element,
     renderer: PropTypes.func,
     now: PropTypes.func,
@@ -86,7 +87,6 @@ export default class Countdown extends React.Component<CountdownProps, Countdown
     onPause: PropTypes.func,
     onTick: PropTypes.func,
     onComplete: PropTypes.func,
-    className: PropTypes.string,
   };
 
   mounted = false;
@@ -274,7 +274,7 @@ export default class Countdown extends React.Component<CountdownProps, Countdown
       );
     }
 
-    const { children, renderer, className } = this.props;
+    const { className, children, renderer } = this.props;
     const renderProps = this.getRenderProps();
 
     if (renderer) {

--- a/src/Countdown.tsx
+++ b/src/Countdown.tsx
@@ -1,5 +1,5 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 
 import LegacyCountdown, { CountdownProps as LegacyCountdownProps } from './LegacyCountdown';
 

--- a/src/LegacyCountdown.tsx
+++ b/src/LegacyCountdown.tsx
@@ -1,5 +1,5 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
 
 export interface CountdownProps {
   readonly count?: number;

